go/buildhelper/helpertypes/types_test.go

```diff
@@ -0,0 +1,63 @@
+package helpertypes
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+func TestSerialization(t *testing.T) {
+	rol := nodecommon.Rollup{
+		Header: &nodecommon.Header{
+			ParentHash:  obscurocommon.GenesisBlock.Hash(),
+			Agg:         common.Address{},
+			Nonce:       2,
+			L1Proof:     obscurocommon.L1RootHash{},
+			State:       "",
+			Height:      0,
+			Withdrawals: nil,
+		},
+		Transactions: nil,
+	}
+
+	serializedRollup := EncodeToString(nodecommon.EncodeRollup(&rol))
+	deserializedRollup := DecodeFromString(serializedRollup)
+	newRollup, err := nodecommon.DecodeRollup(deserializedRollup)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rol.Hash() != newRollup.Hash() {
```

Comment:
@otherview Here and below, any reason we're not sticking some txs in the rollup and checking equality over the whole rollup? Seems easy enough to do and provides more coverage.

Reply:
Good point. I'll add a TODO and come around to it 👍